Make search bar and favourite bar rounded

TBog opened the pull request:
Fix for #822

## Things done:
- make search bar rounded
- make kissbar rounded to match search bar
- make favorites bar rounded (same drawable as search bar)

## Issues:
- Corner radius is the same for both small and large bars

![screenshot_20180226-164045](https://user-images.githubusercontent.com/6465804/36676646-e699c5d2-1b14-11e8-855e-92c1054b0903.png)
![screenshot_20180226-164301](https://user-images.githubusercontent.com/6465804/36676647-e6b94614-1b14-11e8-9fcc-02eb0c2f96c7.png)

licaon-kter commented:
I use transparent, but this looks nice. :)

TBog commented:
@licaon-kter but this also changes the kissbar, so you will have a bar with rounded corners when you press the KISS button

nmitsou commented:
Looks really nice :-) 
Is the history also rounded?

licaon-kter commented:
@TBog A green one, yes...

Neamar commented:
@Tbog pretty neat ;)

Two questions:

* Do we want to put this behind a setting? Or should we enable by default for everyone? Initially I assumed it would be something configurable (especially now that it's easy to add new settings and document them)
* Any chance to increase the corner radius on a large search bar?

TBog commented:
- I will look into making it a setting 
- If we want to have the rounded sides we can use 29dp as radius, it will auto adjust when the height is smaller

licaon-kter commented:
@Neamar Yes and no, try "primary colour Transparent", is `grey`=`transparent`?

Neamar commented:
This is actually hard coded. If you think it's a bad idea can you open a
new issue? Let's keep this thread for the rounded bar

On Mon, Feb 26, 2018, 23:20 Licaon_Kter <notifications@github.com> wrote:

> @Neamar <https://github.com/neamar> Yes and no, try "primary colour
> Transparent", is grey=transparent?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/Neamar/KISS/pull/860#issuecomment-368686906>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAgxDDylPa3ZaH6blOqdRxkf22gttVknks5tYzxBgaJpZM4STWoA>
> .
>

TBog commented:
History without rounded corners￼

iLiwe commented:
Why history is not rounded? Or will there be a setting for it?

TBog commented:
The clipping of the list content to the round corners will work only on LOLLIPOP and above
![screenshot_20180227-133437](https://user-images.githubusercontent.com/6465804/36726756-8f3d2d1a-1bc3-11e8-9f4b-b9350b5aa7bc.png)

TBog commented:
- Added preference toggle for round bars
- Added preference toggle for round history / result list

## Side note:
This PR clashes more and more with PR #847 
Which one is going to get merged first?

Neamar commented on app/src/main/java/fr/neamar/kiss/UiTweaks.java:
```diff
@@ -49,6 +52,40 @@ static void updateThemePrimaryColor(Activity activity) {
         }
     }
 
+    static void applyRoundedCorners(MainActivity mainActivity) {
```
when you rebase, you may want to put  this code in the Interface forwarder (you get access to the mainActivity and SharedPreferences for free)

TBog replied:
I already have a branch that has this rebased, that's what I use on my phone ATM. I just need to double check.

Neamar commented on app/src/main/java/fr/neamar/kiss/UiTweaks.java:
```diff
@@ -49,6 +52,40 @@ static void updateThemePrimaryColor(Activity activity) {
+            mainActivity.findViewById(R.id.main_kissbar).setBackgroundColor(kissGreen);
+
+            // get styled background color
+            int[] attrs = {R.attr.searchBackgroundColor};
```
Thinking out loud, but wouldn't it be easier to understand to create another drawable xml and load it here?

Something like

```xml
<?xml version="1.0" encoding="utf-8"?>
<shape xmlns:android="http://schemas.android.com/apk/res/android"
    android:shape="rectangle">
    <solid android:color="?attr/searchBackgroundColor" />
</shape>
```

Neamar replied:
Or even easier, to keep the default background in the XML to what is used to be (`?attr/searchBackgroundColor`) and then load another background resource only if rounded interface is on?

TBog replied:
I will think about what you said.

Neamar commented on app/src/main/res/values/dimens.xml:
```diff
@@ -19,6 +19,7 @@
 
     <dimen name="bar_height">40dp</dimen>
     <dimen name="large_bar_height">58dp</dimen>
+    <dimen name="corner_radius">29dp</dimen> <!--half of large_bar_height-->
```
Potential caveat: when not using the large search bar, list view will be more rounded than the favorite bar that will be more rounded than the search bar

Neamar commented:
Looks really good :)

TBog commented:
rebase done

Neamar commented:
Cool, thanks!

Did you try simplifying even more as suggested in https://github.com/Neamar/KISS/pull/860#discussion_r171056884?

The manifest specifies `stateNotNeeded` for MainActivity, which means UI is rebuilt from scratch for every onResume. As such, I believe we don't need the custom rectangle drawable, we can just keep the `?attr/searchBackgroundColor` value by default. This also means that there is no need for an `else` branch in the prefs.isRounded() code: if the setting is not enabled by default, then keep UI as it's defined in the XML.

Pros:

* less drawables
* less UI setup work when resuming the app
